apollo-compiler/src/main/kotlin/com/apollographql/apollo/compiler/Inflector.kt

```diff
@@ -9,7 +8,12 @@ import java.util.regex.Pattern
  * The singularization methods were heavily based off rogueweb <link> https://code.google.com/archive/p/rogueweb/source</link>
  */
 
-fun String.singularize(): String {
+fun String.singularize(customMappings: Set<Pair<String, String>>): String {
+  val customMapping = customMappings.lastOrNull { match(it.component1(), this) }
```

Comment:
nit: you can use destructuring here:
```
val (pattern, value) = customMappings....
```

Reply:
Destructuring would work great with `last`, but unfortunately it doesn't work so well with `lastOrNull`.